File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int r = heights.Length - 1;
        int l = 0;
        int maxArea = 0;

        while (l < r) {
            int area = Math.Min(heights[l], heights[r]) * (r - l);
            if (heights[l] <= heights[r]) {
                l++;
            }
            else {
                r--;
            }
            maxArea = Math.Max(area, maxArea);
        }

        return maxArea;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int l = 1;
        int r = Arrays.stream(piles).max().getAsInt();
        int minRate = r;

        while (l <= r)
        {
            int mid = (l + r) / 2;
            int hours = 0;
            for (int pile : piles) {
                hours += Math.ceilDiv(pile, mid);
            }
            if (hours <= h) {
                minRate = mid;
                r = mid - 1;
            }
            else {
                l = mid + 1;
            }
        }
        return minRate;
    }
}
